### PR TITLE
SOL-13 [Integrate] Integrate smart contract function for creating campaign

### PR DESCRIPTION
**Pull request by arden-dave-sun**
## Issue Link
- https://framgiaph.backlog.com/view/SPH_SOLIDITY-13

## Target Page | Component | Function
- /create-campaign

## Pre-condition
- install necessary packages
- setup env files
- Balance in the sepolia network (Sepolia ETH)

## Steps | Commands
1. create .env file in `fe` directory and set the following variables
```
CONTRACT_ADDRESS = "0xb955C4Cee999ACc6D61b061F7907c70dc28eb9F7"
```
2. Create campaign using the form in the FE

## Expected Output
- After clicking `Submit`, submit button should be disabled and metamask will open
- After clicking `Confirm`, there should be message if creation is in progress or failed.
- If the operation failed, there should be an error toast with error message
- After waiting for some time (mining time), if the operation is success, there should be info toast

## Definition of Done
- Create campaign and save it in the blockchain

## Test View Points
- [ ] Add here

## Screenshots | Recordings

https://github.com/framgia/sph-solidity/assets/111718037/fe365627-9922-4b0b-885d-90d8d6673ea8

### General
- [ ] Does the code meet the requirements of the project?
- [ ] Is the code readable and maintainable?
- [ ] Are there any unnecessary or redundant lines of code?
- [ ] Are there any long methods or functions that should be refactored?
- [ ] Are there any performance issues?

### Functionality
- [ ] Are there any logical errors?
- [ ] Are edge cases handled properly?
- [ ] Are there any potential security vulnerabilities?

### Code Style
- [ ] Does the code adhere to the project's coding standards?
- [ ] Are variable and function names descriptive and meaningful?
- [ ] Is there proper use of comments and documentation?
- [ ] Is the code properly formatted? Are indentation, spacing, and formatting consistent?

### Security
- [ ] Is the website secure from common attacks?
- [ ] Are inputs properly validated and sanitized?
- [ ] Are sensitive data and credentials properly protected?

### Testing
- [ ] Are there sufficient test cases?
- [ ] Do the test cases cover all functionality?
- [ ] Are the test cases properly written and documented?

### Additional Comments
- [ ] Are there any other issues or concerns with the code?
- [ ] Are there any suggestions for improvement?
- [ ] Are there any positive aspects of the code that should be highlighted?

### CODE REVIEW

**Review by kenorenzsun — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "hardhat:compile": "yarn workspace smart_contract compile",
     "hardhat:node": "yarn workspace smart_contract run-node",
     "hardhat:deploy-local": "yarn workspace smart_contract deploy-local",
+    "hardhat:deploy-sepolia": "yarn workspace smart_contract deploy-sepolia",
```
Comment (wylenjoan) [inline package.json:19]: Please add this in README in root dir and smart_contract dir

**Review by wylenjoan — DISMISSED** [review]
Approved :white_check_mark:
Review time: 30 mins
Comment: 1
Iteration: 0

Working:
- Creating campaign
Screenshot 2023-06-29 133620

**Review by johnpaul-sun — DISMISSED** [review]
Approved ✅

Review Time: 30 mins
Comments: 0
IT: 0

**Comment by ipilikent** [comment]
Comment
• Problem
- In Goal field if 22 numbers are exceeded, an error will occur
image

• Expected Output
- 10 digit max number

**Comment by ipilikent** [comment]
## Approved PR
- Can create campaign
- Can reject the campaign
- The validation messages are working & displayed
- Can input the field
- The button is clickable

## Proof
<details>

Sol-13-Integrate-Creating-Campai

image

</details>

## Notes
- The created campaign is not displayed yet, but you can see its transaction hash
- The upload image is not working yet

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=637322504
-----
- Total Test time:  1 hr.
- Total Comments: 1
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

**Review by kenorenzsun — APPROVED** [review]

**Review by wylenjoan — APPROVED** [review]

**Review by ipilikent — APPROVED** [review]

**Pull request merged**